### DISCUSSION TITLE
bg-${color}-xxx Template string not working properly

### DISCUSSION THREAD

**Discussion by Dcose** [tailwindlabs/tailwindcss#12377] [Help]
![1](https://github.com/tailwindlabs/tailwindcss/assets/77829706/770e9281-47a0-4480-8cd3-db555df119bc)
![2](https://github.com/tailwindlabs/tailwindcss/assets/77829706/efdece4b-243d-4db1-9095-1cf8febf7b82)
![3](https://github.com/tailwindlabs/tailwindcss/assets/77829706/27a85357-577a-4dd1-b190-04220a2565e0)
![4](https://github.com/tailwindlabs/tailwindcss/assets/77829706/11abaac6-f034-459c-8928-53243c760efe)
Case 1: I can see in the console that the class bg-color-xxx has been loaded, but the corresponding effect is not displayed.
Case 2: When I change `after:bg-${color}-500` to `after:bg-pink-500`
and then use `<NavLink href="#" color="pink">Test</NavLink>` will be displayed successfully.
I'm very confused, I hope you know how to solve it, thank you very much！！！

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```

**Reply by Dcose** [reply]
thanks！
